exporter/datadogexporter/internal/metrics/transform.go

```diff
@@ -0,0 +1,187 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metrics"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/metricstransformprocessor"
+)
+
+// TransformFunc takes a set of metrics and transforms them.
+type TransformFunc func(ctx context.Context, md pmetric.Metrics) error
+
+// NewInfraTransformFunc returns a new TransformFunc which converts OpenTelemetry system and process metrics
+// to Datadog compatible ones.
+func NewInfraTransformFunc(ctx context.Context, set component.ExporterCreateSettings) (TransformFunc, error) {
+	cid := config.NewComponentIDWithName(config.MetricsDataType, "dd-infra-metricstransformer")
+	tcfg := &metricstransformprocessor.Config{
+		ProcessorSettings: config.NewProcessorSettings(cid),
+		Transforms:        metricTransforms.Transforms,
+	}
+	pset := component.ProcessorCreateSettings(set)
+	noop, err := consumer.NewMetrics(
+		consumer.ConsumeMetricsFunc(
+			func(_ context.Context, _ pmetric.Metrics) error { return nil },
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+	proc, err := metricstransformprocessor.NewFactory().CreateMetricsProcessor(ctx, pset, tcfg, noop)
+	if err != nil {
+		return nil, err
+	}
+	return proc.ConsumeMetrics, err
+}
+
+func init() {
```

Comment:
nit: I would rather have one `metricTransforms` per `TransformFunc` and do this initialization inside `NewInfraTransformFunc` instead of in the `init` function. 

The reason is that if we ever add a config option that allows modifying the transform behavior in any way we won't have the configuration available in the `init` function so I would rather do it the 'proper' way now. Having multiple Datadog exporters e.g. for dual shipping would also be problematic, but that's a more niche case.

Reply:
I agree. The only reason I did this was because of [this comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13980#discussion_r968460989). I am not a fan of `init()` functions. They are just looking for trouble.

However, if I move this into the constructor, then we have to run a for loop over all of the metrics on every call of `shouldPrepend` to identify our native metrics. Are we ok with that? Frankly I'd prefer it too.